step-automation-packages/step-automation-packages-manager/src/main/java/step/automation/packages/AutomationPackageManager.java

```diff
@@ -259,35 +266,89 @@ public PackageUpdateResult createOrUpdateAutomationPackage(boolean allowUpdate,
             ObjectEnricher enricherForIncludedEntities = ObjectEnricherComposer.compose(Arrays.asList(enricher, new AutomationPackageLinkEnricher(newPackage.getId().toString())));
             fillStaging(staging, packageContent, newPackage, oldPackage, enricherForIncludedEntities, automationPackageArchive);
 
+            // save automation package metadata
+            if (oldPackage == null || !async) {
```

Comment:
Would add some debug/info logging here to diagnose potential issues with locking